lib/lobby.ts

```diff
@@ -0,0 +1,184 @@
+import { getClient } from '.';
+import { createSpotifyPlaylist } from '../utils/createSpotifyPlaylist';
+import { DbItem, IDbItem } from './db-item';
+import { COLLECTION } from './private/enums';
+import { User } from './user';
+
+type DatabaseEntry = Omit<ILobby, 'collectionName'>;
+type ClientResponse = Omit<DatabaseEntry, 'users' | 'uri'>;
+
+export interface LobbyProps {
+  /**
+  * The spotify id of playlist
+  */
+  readonly spotifyPlaylistId?: string;
+
+  /**
+   * The manager of a lobby
+   */
+  readonly managerId: string;
+
+  /**
+   * The participants in a lobby
+   */
+  readonly participants: string[];
+
+  /**
+   * The theme of a lobby
+   */
+  readonly theme: string;
+
+  /**
+   * The playlist name
+   */
+  readonly name: string;
+
+  /**
+   * The list of song ids in the playlist
+   */
+  readonly songIds: string[];
+}
+
+export interface ILobby extends LobbyProps, IDbItem {
+}
+
+/**
+ * The class containing a user and their data
+ */
+export class Lobby extends DbItem implements ILobby {
+  /**
+   * A static function to query for a user from their id
+   *
+   * @returns a user object if the id exists in the database
+   */
+  public static async fromId(id: string): Promise<Lobby | null> {
+    const client = await getClient();
+    const document = await client.findDbItem(COLLECTION.LOBBIES, id);
+    if (!document) return null;
+    const content: DatabaseEntry = document.getContent() as DatabaseEntry;
+    return new Lobby(id, content, document.key ?? null);
+  }
+
+  /**
+   * A static function to create a new Lobby and corresponding Spotify playlist
+   *
+   * @returns a newly created Lobby object
+   */
+  public static async create(props: LobbyProps, key : string | null = null) : Promise<Lobby | null> {
```

Comment:
Lets make a new interface for this... i think when we create a lobby, we don't need any songId, participants, or spotifyPlaylistId